### ISSUE TITLE
Reflowing of text

### ISSUE THREAD

**Issue by jrbowden** [daisy/ebraille#4]
As a braille display user or braille organisation, I want to optimally use the available braille line length when reading or embossing a braille document, which may have been produced by an organisation with a different default braille line length to my own.

*Detail:*
If my braille display or embosser has a different braille line length to the organisation who produced a document, ideally the braille in ordinary document elements should be "reflowed" to make optimal use of the available line length I have - taking into account any special formatting requirements.

What should not happen is: 
- only a portion of the line length is used (if the original document had a shorter line length to my own), or
- the document renders as full line, part line, (if the original line length is longer than my own).

Currently, braille files contain hard line breaks according to the original line length, set by the originating organisation, and these cannot easily be modified for a different target line length.

*Proposal:*
The braille file should be organised according to document structural elements, rather than by hard lines and pages fixed to a particular page size.
This would enable ordinary text elements to be flowed onto whatever line length is available.

**Comment by clapierre** [comment]
During our discussion it was pointed out that this feature can not be done with any existing braille first format currently.

**Comment by franciscoONCE** [comment]
This requirement can also be mapped with EPUB's ability to reflow text.

**Comment by avneeshsingh** [comment]
It would be good to map this capability to XHTML / HTML instead of EPUB, because EPUB is the container of XHTML. XHTML / HTML standard allow use of these features on the web as well as in EPUB and other packaging formats which would be used in the future.

**Comment by ways2read** [comment]
Also true for the DAISY format

**Comment by mattgarrish** [comment]
This sounds more like a device issue. All the publishing formats are reflowable by default, with the exception of certain elements designed for preformatted text.

It sounds like it could be complicated by the `pre` element or CSS that is intended to retain line lengths for specific reasons, but that's still not an authoring issue.

**Comment by Menelion** [comment]
Sorry, but what about hyphenation, especially in different languages? If the format is to be used not only for Braille displays, but also for embossers, this must also be taken into account. I.e., if a document was prepared by an organization for a 32-cell line and properly hyphenated, and then you want to view it on a standard 40-cell Braille display, what should happen to the hyphenation already in place?

**Comment by bertfrees** [comment]
@Menelion Good point. The hyphenation already in place should obviously be dropped in this case.

Technically this could be done as follows: The hyphens may be represented by special unicode characters (U+00AD), together with the replacement character / braille pattern stored in CSS (the 'hyphenation-character' property). The soft hyphen is only replaced by the real hyphen when it occurs before a line break, otherwise it is invisible.

There is one catch with this method: in some languages hyphenation can alter the spelling.  I don't know of a way to deal with this, except for hyphenating on the fly (by the device).

**Comment by jrbowden** [comment]
I recently added use case #51. Optional hyphens and no-break spaces are I think required to help with reflowing of text.

**Comment by Menelion** [comment]
As I was reading issues gradually, I didn't see #51 at that point, it's probably a better place for this discussion.
